app/models/image.rb

```diff
@@ -15,6 +15,23 @@ class Image < ActiveRecord::Base
 
   scope :donor_images, ->(donor_id) { joins(item: [:offer]).where(offers: { created_by_id: donor_id }) }
 
+  # Live update rules
+  after_save :push_changes
+  after_destroy :push_changes
+  push_targets do |record|
+    package = record.imageable if record.imageable_type == "Package"
+    channels = []
+    if record.offer
+      channels << Channel.private_channels_for(record.offer.created_by_id, DONOR_APP)
+      channels << Channel::STAFF_CHANNEL
+    end
+    if package
+      channels << Channel::STOCK_CHANNEL if package.inventory_number.present?
+      channels << Channel::BROWSE_CHANNEL if (package.allow_web_publish || package.allow_web_publish_was)
+    end
+    channels
+  end
```

Comment:
```suggestion
  push_targets do |record|
    package = record.imageable if record.imageable_type == "Package"
    channels = []
    channels << Channel.private_channels_for(record.offer.created_by_id, DONOR_APP)  if record.offer
    channels << Channel::STAFF_CHANNEL if record.offer
    channels << Channel::STOCK_CHANNEL if package&.inventory_number.present?
    channels << Channel::BROWSE_CHANNEL if (package&.allow_web_publish || package&.allow_web_publish_was)
    channels
  end
```

I don't really mind either way. Just an alternative for fun :)